### DISCUSSION TITLE
Create more detailed documentation for all Maven Plugins - https://maven.apache.org/plugins/

### DISCUSSION THREAD

**Discussion by khmarbaise** [support-and-care/maven-support-and-care#12] [Ideas for Funding]
* How are those plugins used?
* In which circumstances are they used?
* How could they be configured? 

* For example (no particular order):
  * maven-compiler-plugin
    * How to be used? 
    * Changing the JDK?
    * Configure Warnings/Deprecations
  * maven-jar-plugin
    * Test JAR 
  * maven-surefire-plugin/maven-failsafe-plugin
    * How to be used in a project? https://www.youtube.com/@khmarbaise (examples)

Related to #3 #5

**Comment by sparsick** [comment]
Another idea: create an awesome maven plugin list where 3rd party plugins can be listed

**Comment by Bukama** [comment]
> Changing the JDK?
This might be connected to toolchains plugin.

> How are those plugins used?
> In which circumstances are they used?

During that also: When to configure something else than default (most defaults are good for 99% of the users) and should not be "fix" by the user (for case the default changes)

**Comment by ascheman** [comment]
This discussion has been converted to Epic #58 for tracking.

Closing this discussion.
